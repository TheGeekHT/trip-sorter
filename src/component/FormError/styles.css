.errors{
    padding: 0 22px;
    margin: 0;
}

.error{
    margin: 5px 0;
    padding: 0;
    list-style: disc;
    color: #a94442;
}