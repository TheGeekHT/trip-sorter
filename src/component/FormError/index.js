// @flow

import React from 'react'
import { Panel } from 'react-bootstrap'

import Icon from '../Icon'

import './styles.css'

type Props = {
  title: string,
  errors?: Array<string>,
}


const FormError = ({title, errors}: Props) => {
  const titleWrapper = <h2><Icon type="glyphicon glyphicon-exclamation-sign" /> <strong>{title}</strong></h2>;

  return (
    <Panel header={titleWrapper} bsStyle="danger">
      <ul className="errors">
        {errors ? errors.map((err, i) => <li className="error" key={i}>{err}</li>) : '' }
      </ul>
    </Panel>
  )
};

export default FormError
