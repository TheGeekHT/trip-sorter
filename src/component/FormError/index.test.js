import React from 'react'
import renderer from 'react-test-renderer'
import FormError from './'

const errors = [
  'error 1',
  'error 2',
];

describe('form error', () => {
  test('empty should render without crashing', () => {
    const component = renderer.create(
      <FormError />
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render without crashing', () => {
    const component = renderer.create(
      <FormError errors={errors} />
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});