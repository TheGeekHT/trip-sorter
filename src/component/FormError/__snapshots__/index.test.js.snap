exports[`form error empty should render without crashing 1`] = `
<div
  className="panel panel-danger"
  id={undefined}>
  <div
    className="panel-heading">
    <h2
      className="panel-title">
      <i
        className="glyphicon glyphicon-exclamation-sign" />
       
      <strong />
    </h2>
  </div>
  <div
    className="panel-body">
    <ul
      className="errors">
      
    </ul>
  </div>
</div>
`;

exports[`form error should render without crashing 1`] = `
<div
  className="panel panel-danger"
  id={undefined}>
  <div
    className="panel-heading">
    <h2
      className="panel-title">
      <i
        className="glyphicon glyphicon-exclamation-sign" />
       
      <strong />
    </h2>
  </div>
  <div
    className="panel-body">
    <ul
      className="errors">
      <li
        className="error">
        error 1
      </li>
      <li
        className="error">
        error 2
      </li>
    </ul>
  </div>
</div>
`;
